File: Udemy/Algoritmos e Logica de Programacao/C/Matrizes/ProblemaCadaLinha.c
/*
Problema "cada_linha"
Ler um inteiro N e uma matriz quadrada de ordem N (máximo = 10). Mostrar qual o maior elemento
de cada linha. Suponha não haver empates.
*/

#include<stdio.h>
#include<stdlib.h>
int main (){
	
	printf("Qual a ordem da matriz? ");
	int n = 0;
	scanf("%d",&n);
	int i=0;
	int j=0;
	int novo[n];
	int vet[n][n];
	int maior = 0;
	for(i=0;i<n;i++){
		for(j=0;j<n;j++){
			printf("Elemento [%d,%d]: ",i,j);
			scanf("%d",&vet[i][j]);
		}
	}
	
	
	
	for(i=0;i<n;i++){
		maior = vet[i][0];
		for(j=0;j<n;j++){
			if(vet[i][j]>maior){
				maior = vet[i][j];
			}
		}
		novo[i] = maior;
	}
	
	printf("MAIOR ELEMENTO DE CADA LINHA: \n");
	
	for(i=0;i<n;i++){
		printf("%d\n",novo[i]);
	}

    return 0;
}

File: Algoritmos e Logica de Programacao/Java/ExerciciosJavaMatrizes/src/application/ProblemaCadaLinha.java
package application;

import java.util.Locale;
import java.util.Scanner;

public class ProblemaCadaLinha {

	public static void main(String[] args) {
		/*
		 * Problema "cada_linha"
Ler um inteiro N e uma matriz quadrada de ordem N (máximo = 10). Mostrar qual o maior elemento
de cada linha. Suponha não haver empates.
		 * */
		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Qual a ordem da matriz ? ");
        int n = sc.nextInt();
        int[][] mat = new int[n][n];
        int[] maiorLinha = new int[n];
        int maior = 0;
        for (int i = 0; i < n; i++)
        {
            for(int j = 0; j < n; j++)
            {
                System.out.print("Elemento [" + i + "," + j + "]: ");
                mat[i][j] = sc.nextInt();
            }
        }

        System.out.println("MAIOR ELEMENTO DE CADA LINHA: ");

        for (int i = 0; i < n; i++)
        {
            maior = mat[i][0];
            for (int j = 0; j < n; j++)
            {
                if (mat[i][j] > maior)
                {
                    maior = mat[i][j];
                }
            }
            maiorLinha[i] = maior;
        }

        for (int i = 0; i < n; i++)
        {
            System.out.println(maiorLinha[i]);
        }
		
		sc.close();
	}

}
